unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -2281,8 +2307,27 @@ importRemoteShareBranch ReadShareRemoteNamespace {server, repo, path} = do
   mapLeft Output.ShareError <$> do
     let shareFlavoredPath = Share.Path (repo Nel.:| coerce @[NameSegment] @[Text] (Path.toList path))
     LoopState.Env {authHTTPClient, codebase = codebase@Codebase {connection}} <- ask
-    liftIO (Share.pull authHTTPClient baseURL connection shareFlavoredPath) >>= \case
-      Left e -> pure (Left (Output.ShareErrorPull e))
+    let pull :: IO (Either Share.PullError CausalHash)
```

Comment:
Would you mind abstracting the callback here like you did with the `upload` version? I find it much easier to follow the flow of the other one because it doesn't interleave the important bits with the progress bits, plus then the two will be consistent with each other 😄